samples/match/match_enum_else_binding.jakt

```diff
@@ -0,0 +1,26 @@
+/// Expect:
+/// - output: "1\n2\n"
+
+enum Foo {
+    A(x: i64)
+    B(x: i64)
+    C(x: i64)
+}
+
+function main() {
+    let foo = Foo::A(x: 1)
+
+    match foo {
+        else(x) => {
+            println("{}", x)
+        }
+    }
+
+    let bar = Foo::B(x: 2)
+
+    match bar {
+        else(x) => {
```

Comment:
can we add another test where we match some, but not all, of the variants and then use `else(x)` to catch the rest of the cases and require that the remainder have the common field?

Reply:
Sure, I added it to the example and added a test for the failure case where a variant is missing the common field and a test for failing when attempting to use it with a generic match.

Reply:
I meant something more like this:

```
enum Foo {
    A(x: i64)
    B(x: i64)
    C(x: i64)
}

function main() {
    let a = Foo::A(x: 1)

    match a {
        A(x) => println("I found an A {}", x)
        else(x) => println("Just {}", x)
    }
}
```

Reply:
Added that case as well, reversed the cases so that it also tests that it doesn't match A in the else case when it appears first.